### PR TITLE
refactor: Api trace graph Restructuring

### PR DESCRIPTION
**Pull request by rish691**
No logical change

### CODE REVIEW

**Review by skjindal93 — APPROVED** [review]
Any motivation behind the refactor? (_apart from the event id change to ByteBuffer from String_)

**Review by JBAhire — APPROVED** [review]

**Comment by rish691** [comment]
> Any motivation behind the refactor? (_apart from the event id change to ByteBuffer from String_)

Was working on [this](https://github.com/hypertrace/hypertrace-ingester/pull/144) change to move some logic from view-generators to ApiTraceGraph. So, just moved the basic refactoring into this pr.

File: hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/ApiTraceGraph.java [hunk 1]
```diff
@@ -290,23 +280,24 @@ private void buildApiNodeToIndexMap(List<ApiNode<Event>> apiNodes) {
     return Optional.empty();
   }
 
-  private Table<Integer, Integer, Edge> buildTraceEdgeTable() {
-    Table<Integer, Integer, Edge> traceEdgeTable = HashBasedTable.create();
+  private void buildApiNodeToIndexMap() {
+    for (int i = 0; i < apiNodeList.size(); i++) {
```
Comment (kotharironak) [inline hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/ApiTraceGraph.java:284]: can we convert this for() to apiNodeList.foreach ?

**Review by kotharironak — APPROVED** [review]
LGTM

**Comment by kotharironak** [comment]
@rish691 can we handle snyk issues as part of this PR for ingester?

**Comment by rish691** [comment]
> @rish691 can we handle snyk issues as part of this PR for ingester?

I see that the dependency is already on the asked version 
```
|    +--- io.netty:netty-codec-http2:4.1.52.Final -> 4.1.59.Final
|    |    +--- io.netty:netty-common:4.1.59.Final
|    |    +--- io.netty:netty-buffer:4.1.59.Final
|    |    |    \--- io.netty:netty-common:4.1.59.Final
|    |    +--- io.netty:netty-transport:4.1.59.Final
|    |    |    +--- io.netty:netty-common:4.1.59.Final
|    |    |    +--- io.netty:netty-buffer:4.1.59.Final (*)
|    |    |    \--- io.netty:netty-resolver:4.1.59.Final
|    |    |         \--- io.netty:netty-common:4.1.59.Final
|    |    +--- io.netty:netty-codec:4.1.59.Final
|    |    |    +--- io.netty:netty-common:4.1.59.Final
|    |    |    +--- io.netty:netty-buffer:4.1.59.Final (*)
|    |    |    \--- io.netty:netty-transport:4.1.59.Final (*)
|    |    +--- io.netty:netty-handler:4.1.59.Final
|    |    |    +--- io.netty:netty-common:4.1.59.Final
|    |    |    +--- io.netty:netty-resolver:4.1.59.Final (*)
|    |    |    +--- io.netty:netty-buffer:4.1.59.Final (*)
|    |    |    +--- io.netty:netty-transport:4.1.59.Final (*)
|    |    |    \--- io.netty:netty-codec:4.1.59.Final (*)
|    |    \--- io.netty:netty-codec-http:4.1.59.Final
|    |         +--- io.netty:netty-common:4.1.59.Final
|    |         +--- io.netty:netty-buffer:4.1.59.Final (*)
|    |         +--- io.netty:netty-transport:4.1.59.Final (*)
|    |         +--- io.netty:netty-codec:4.1.59.Final (*)
|    |         \--- io.netty:netty-handler:4.1.59.Final (*)
|    +--- io.netty:netty-handler-proxy:4.1.52.Final -> 4.1.59.Final
|    |    +--- io.netty:netty-common:4.1.59.Final
|    |    +--- io.netty:netty-buffer:4.1.59.Final (*)
|    |    +--- io.netty:netty-transport:4.1.59.Final (*)
|    |    +--- io.netty:netty-codec:4.1.59.Final (*)
|    |    +--- io.netty:netty-codec-socks:4.1.59.Final
|    |    |    +--- io.netty:netty-common:4.1.59.Final
|    |    |    +--- io.netty:netty-buffer:4.1.59.Final (*)
|    |    |    +--- io.netty:netty-transport:4.1.59.Final (*)
|    |    |    \--- io.netty:netty-codec:4.1.59.Final (*)
|    |    \--- io.netty:netty-codec-http:4.1.59.Final (*)
```

File: hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/ApiTraceGraph.java (hunk 1 above)
Comment (rish691) [reply hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/ApiTraceGraph.java:284]: I will do it here https://github.com/hypertrace/hypertrace-ingester/pull/144

Comment (rish691) [reply hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/ApiTraceGraph.java:284]: Actually we need the index here.

**Pull request merged**